Neo4j.Driver/Neo4j.Driver.Tests/Result/PeekingEnumeratorTests.cs

```diff
@@ -115,15 +115,32 @@ public void ShouldConsumeAllItemsAfterDiscard()
         public class DisposeMethod
         {
             [Fact]
-            public void ShouldNotDiscardAfterDispose()
+            public void ShouldDiscardAfterDispose()
             {
                 var list = new List<string> { "1", "2", "3" }.GetEnumerator();
                 var enumerator = new PeekingEnumerator<string>(list);
-                enumerator.MoveNext().Should().BeTrue();
-                enumerator.Consume();
+                enumerator.Dispose();
                 enumerator.MoveNext().Should().BeFalse();
                 enumerator.Peek().Should().BeNull();
-                enumerator.Position.Should().Be(3);
+                enumerator.Position.Should().Be(-1);
+            }
+
+            [Fact]
+            public void PeakShouldReturnDefaultAfterDispose()

```

Comment:
I would suggest in this PR we just focus on fixing the compiling warning, and then leave the changes to the result API to another PR, as we currently have disagreement on the use of Dispose and enumerator, WDYT?